crates/storage-query-datafusion/src/remote_query_scanner_manager.rs

```diff
@@ -79,18 +79,25 @@ pub trait PartitionLocator: Send + Sync + 'static {
 #[derive(Clone)]
 struct MetadataAwarePartitionLocator {
     partition_routing: PartitionRouting,
+    metadata: Metadata,
 }
 
-pub fn create_partition_locator(partition_routing: PartitionRouting) -> Arc<dyn PartitionLocator> {
-    Arc::new(MetadataAwarePartitionLocator { partition_routing })
+pub fn create_partition_locator(
+    partition_routing: PartitionRouting,
+    metadata: Metadata,
+) -> Arc<dyn PartitionLocator> {
+    Arc::new(MetadataAwarePartitionLocator {
+        partition_routing,
+        metadata,
+    })
 }
 
 impl PartitionLocator for MetadataAwarePartitionLocator {
     fn get_partition_target_node(
         &self,
         partition_id: PartitionId,
     ) -> anyhow::Result<PartitionLocation> {
-        let my_node_id = my_node_id();
```

Comment:
Why does this commit introduces the `metadata`  in favor of `my_node_id()`? Is it because of performance considerations when accessing a task local field?

Reply:
it's because this code gets executed from a tokio-spawned task initiated by data-fusion, unfortunately we can't decorate the future that it spawns.